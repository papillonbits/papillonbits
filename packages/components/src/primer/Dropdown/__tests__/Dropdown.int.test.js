import { items, regular, inactive } from './Dropdown.int.story'

jest.mock('@papillonbits/library/array', () => ({
  getIndexItems: () => {},
  getIndexItemsWithSelected: () => {},
}))

jest.mock('@papillonbits/library/hooks', () => ({
  useRef: () => {},
}))

const libraryArrayMockObject = require('@papillonbits/library/array')
const libraryHooksMockObject = require('@papillonbits/library/hooks')

describe('<Dropdown />', () => {
  const indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
    { ...items[2], index: 2 },
    { ...items[3], index: 3 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
    { ...items[2], index: 2, isSelected: false },
    { ...items[3], index: 3, isSelected: false },
  ]

  beforeEach(() => {
    jest.spyOn(libraryArrayMockObject, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryArrayMockObject, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)

    jest.spyOn(libraryHooksMockObject, 'useRef').mockImplementation(() => ({
      current: {
        removeAttribute: () => {},
      },
    }))
  })

  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match regular()', () => {
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match inacive()', () => {
      expect(global.renderToJSON(inactive())).toMatchSnapshot()
    })
  })
})
