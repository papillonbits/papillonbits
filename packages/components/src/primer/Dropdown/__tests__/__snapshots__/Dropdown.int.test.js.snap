// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Render must match inacive() 1`] = `
<div>
  <div>
    <details
      class="dropdown details-reset details-overlay d-inline-block dropdown-inactive"
    >
      <summary
        aria-haspopup="true"
        class="btn"
      >
        <div
          class="d-flex flex-justify-between flex-items-center dropdown-inactive"
        >
          <div>
            Dropdown
          </div>
           
          <div
            class="dropdown-caret"
          />
        </div>
      </summary>
      <ul
        class="dropdown-menu dropdown-menu-se width-full"
      >
        <li>
          <a
            aria-current="page"
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 1
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 2
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 3
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 4
          </a>
        </li>
      </ul>
    </details>
  </div>
</div>
`;

exports[`<Dropdown /> Render must match regular() 1`] = `
<div>
  <div>
    <details
      class="dropdown details-reset details-overlay d-inline-block"
    >
      <summary
        aria-haspopup="true"
        class="btn"
      >
        <div
          class="d-flex flex-justify-between flex-items-center"
        >
          <div>
            Dropdown
          </div>
           
          <div
            class="dropdown-caret"
          />
        </div>
      </summary>
      <ul
        class="dropdown-menu dropdown-menu-se width-full"
      >
        <li>
          <a
            aria-current="page"
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 1
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 2
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 3
          </a>
        </li>
        <li>
          <a
            class="dropdown-item"
            href="#url"
          >
            Dropdown Item 4
          </a>
        </li>
      </ul>
    </details>
  </div>
</div>
`;
